From 5a74f7f7816d3574df4f87f2d9f4b3af9105f859 Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Sun, 17 Aug 2025 09:14:36 -0700
Subject: [PATCH] CMake: Fix Fast-Math with Pre-built AMReX

The fast-math flags were not added if AMReX was pre-built
without fast-math. This fixes it.
---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ae0a27e..cd6c1170 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,6 +185,11 @@ if(ImpactX_UNITY_BUILD)
     )
 endif()
 
+# Fast-Math
+if(ImpactX_FASTMATH)
+    target_link_libraries(lib PUBLIC AMReX::Flags_FASTMATH)
+endif()
+
 # executable application
 #   note: we currently avoid a dependency on a core library
 #         for simpler usage, but could make this an option
-- 
2.49.0

